testsuite/integration/multinode/src/test/java/org/jboss/as/test/multinode/security/AbstractSecurityPropagationTestCase.java

```diff
@@ -0,0 +1,235 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jboss.as.test.multinode.security;
+
+import java.io.InputStream;
+
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.TargetsContainer;
+import org.jboss.as.arquillian.api.ContainerResource;
+import org.jboss.as.arquillian.container.ManagementClient;
+import org.jboss.as.test.multinode.security.api.EJBAction;
+import org.jboss.as.test.multinode.security.api.EJBRequest;
+import org.jboss.as.test.multinode.security.api.RemoteEJBConfig;
+import org.jboss.as.test.multinode.security.api.Results;
+import org.jboss.as.test.multinode.security.api.SecuritySLSBRemote;
+import org.jboss.as.test.multinode.security.api.ServerConfigs;
+import org.jboss.as.test.multinode.security.api.TestConfig;
+import org.jboss.as.test.multinode.security.config.Deployments;
+import org.jboss.as.test.multinode.security.util.EJBUtil;
+import org.jboss.as.test.multinode.security.util.URLUtil;
+import org.jboss.logging.Logger;
+import org.jboss.shrinkwrap.api.Archive;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * @author bmaxwell
+ *
+ */
+public abstract class AbstractSecurityPropagationTestCase {
+
+    protected static final Logger log = Logger.getLogger(AbstractSecurityPropagationTestCase.class);
+
+    @ContainerResource("multinode-server")
+    protected ManagementClient serverServer;
+
+    @ContainerResource("multinode-client")
+    protected ManagementClient clientServer;
+
+    protected ServerConfigs servers;
+
+    @Before
+    public void init() {
+        servers = new ServerConfigs(clientServer, TestConfig.MULTINODE_CLIENT, serverServer, TestConfig.MULTINODE_SERVER);
+    }
+
+    @Deployment(testable = false, name = "JAR1")
+    @TargetsContainer(TestConfig.MULTINODE_SERVER)
+    public static Archive<?> createEjbDeployment1() {
+        return Deployments.createEjbDeployment(TestConfig.SECURITY_EJB1);
+    }
+
+    @Deployment(testable = false, name = "JAR2")
+    @TargetsContainer(TestConfig.MULTINODE_CLIENT)
+    public static Archive<?> createEjbDeployment2() {
+        return Deployments.createEjbDeployment(TestConfig.SECURITY_EJB1);
+    }
+
+    @Deployment(testable = false, name = "WAR", order = 2, managed = true)
+    @TargetsContainer("multinode-client")
+    public static Archive<?> createWarDeploymentUsingRemoteConnections() {
+        return Deployments.createWarDeployment("", false, TestConfig.SERVLET_1);
+    }
+
+    @Test
+    public void testSecurityPropagationSecuredWebClientToRemoteEJB() throws Exception {
+        // unit test (WEB_CREDENTIAL) -> servlet1 (Client Server) (NO Credential) -> ejb1 (Server Server)
```

Comment:
@mchoma Each test added by @bmaxwell are following by a brief comment describing the test case (as shown above).  Based on this, it seems that maybe this is already by the [testForwardedAuthenticationIdentityDoesNotExistOnIntermediateServer](https://github.com/wildfly/wildfly/blob/master/testsuite/integration/manualmode/src/test/java/org/wildfly/test/manual/elytron/seccontext/ServerChainSecurityContextPropagationTestCase.java#L208) (albeit not using EJB). 

(I'm honestly out of my depth here, I would like to help more, but I'm not sure if I can really assess how the tests relates or not to each other)

Reply:
What do you mean by "not using EJB" ?